modules/auxiliary/admin/mysql/mysql_enum.rb

```diff
@@ -52,7 +54,15 @@ def report_cred(opts)
   end
 
   def run
-    return if not mysql_login_datastore
+    # If we have a session make use it
+    if session
+      print_status("Using existing session #{session.sid}")
+      self.mysql_conn = session.client
```

Comment:
Not a blocker; and maybe one for a future PR. But maybe we should move all of this logic into the new optional session MySQL mixin:

```
def assign_mysql_conn
  print_status("Using existing session #{session.sid}")
  self.mysql_conn = session.client
  self.conn = session.client.conn # This might be required
end
```

Reply:
Probably makes sense to explore this approach after the new mixins are available: https://github.com/rapid7/metasploit-framework/pull/18770

Reply:
@dwelch-r7

Reply:
Yup totally agree, thinking a new PR that handles this across the sessions makes the most sense